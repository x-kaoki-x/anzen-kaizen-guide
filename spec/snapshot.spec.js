const path = require('path');

// テストの実行
// npx jest ./spec/snapshot.spec.js
describe('TODOアプリ', () => {
  beforeEach(async () => {
    await page.goto(`file://${path.resolve(__dirname, '../index.html')}`);
  });

  test('初期表示', async () => {
    expect(await page.content()).toMatchSnapshot();
  });

  test('タスクの追加', async () => {
    await page.click('#addTodo');
    await page.click('#addTodo');
    expect(await page.content()).toMatchSnapshot();
  });

  test('タスクの入力', async () => {
    await page.click('#addTodo');
    await page.click('#addTodo');
    await page.type('.todo:nth-child(1) input', 'サンプルタスク1');
    await page.type('.todo:nth-child(2) input', 'サンプルタスク2');
    expect(await page.content()).toMatchSnapshot();
  });

  test('タスクの削除', async () => {
    await page.click('#addTodo');
    await page.click('#addTodo');
    await page.type('.todo:nth-child(1) input', 'サンプルタスク1');
    await page.type('.todo:nth-child(2) input', 'サンプルタスク2');
    await page.click('.todo:nth-child(1) .delete');
    await page.click('.todo:nth-child(1) .delete');
    expect(await page.content()).toMatchSnapshot();
  });
});
