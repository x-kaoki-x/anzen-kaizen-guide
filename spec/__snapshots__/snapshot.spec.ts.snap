// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TODOアプリ タスクの入力 1`] = `
"<!DOCTYPE html><html><head>
    <meta charset=\\"utf-8\\">
    <link rel=\\"stylesheet\\" href=\\"./css/style.css\\">
    <title>TODO</title>
  </head>
  <body>
    <button id=\\"addTodo\\">タスクを追加する</button>
    <div id=\\"todoList\\" style=\\"\\"><div class=\\"todo\\"><input type=\\"text\\"><button class=\\"delete\\">削除</button></div><div class=\\"todo\\"><input type=\\"text\\"><button class=\\"delete\\">削除</button></div></div>
    <div id=\\"todoEmpty\\" style=\\"display: none;\\">タスクがありません</div>
    <div>
      <span id=\\"nextTodo\\">次のTODO: サンプルタスク1</span>
      <span id=\\"todoCount\\">(全2件)</span>
    </div>
    <script src=\\"./js/jquery-3.3.1.min.js\\"></script>
    <script src=\\"./js/script.js\\"></script>
   

</body></html>"
`;

exports[`TODOアプリ タスクの削除 1`] = `
"<!DOCTYPE html><html><head>
    <meta charset=\\"utf-8\\">
    <link rel=\\"stylesheet\\" href=\\"./css/style.css\\">
    <title>TODO</title>
  </head>
  <body>
    <button id=\\"addTodo\\">タスクを追加する</button>
    <div id=\\"todoList\\" style=\\"display: none;\\"></div>
    <div id=\\"todoEmpty\\" style=\\"\\">タスクがありません</div>
    <div>
      <span id=\\"nextTodo\\">次のTODO: (未登録)</span>
      <span id=\\"todoCount\\">(全0件)</span>
    </div>
    <script src=\\"./js/jquery-3.3.1.min.js\\"></script>
    <script src=\\"./js/script.js\\"></script>
   

</body></html>"
`;

exports[`TODOアプリ タスクの追加 1`] = `
"<!DOCTYPE html><html><head>
    <meta charset=\\"utf-8\\">
    <link rel=\\"stylesheet\\" href=\\"./css/style.css\\">
    <title>TODO</title>
  </head>
  <body>
    <button id=\\"addTodo\\">タスクを追加する</button>
    <div id=\\"todoList\\" style=\\"\\"><div class=\\"todo\\"><input type=\\"text\\"><button class=\\"delete\\">削除</button></div><div class=\\"todo\\"><input type=\\"text\\"><button class=\\"delete\\">削除</button></div></div>
    <div id=\\"todoEmpty\\" style=\\"display: none;\\">タスクがありません</div>
    <div>
      <span id=\\"nextTodo\\">次のTODO: </span>
      <span id=\\"todoCount\\">(全2件)</span>
    </div>
    <script src=\\"./js/jquery-3.3.1.min.js\\"></script>
    <script src=\\"./js/script.js\\"></script>
   

</body></html>"
`;

exports[`TODOアプリ 初期表示 1`] = `
"<!DOCTYPE html><html><head>
    <meta charset=\\"utf-8\\">
    <link rel=\\"stylesheet\\" href=\\"./css/style.css\\">
    <title>TODO</title>
  </head>
  <body>
    <button id=\\"addTodo\\">タスクを追加する</button>
    <div id=\\"todoList\\" style=\\"display: none;\\"></div>
    <div id=\\"todoEmpty\\">タスクがありません</div>
    <div>
      <span id=\\"nextTodo\\">次のTODO: (未登録)</span>
      <span id=\\"todoCount\\">(全0件)</span>
    </div>
    <script src=\\"./js/jquery-3.3.1.min.js\\"></script>
    <script src=\\"./js/script.js\\"></script>
   

</body></html>"
`;
